mail/api.py

```diff
@@ -377,15 +378,15 @@ def get_mail_vars(emails):
         generator of dict:
             A dictionary of template variables which includes email so we can tell who is who
     """
-    queryset = User.objects.filter(email__in=emails).values(
-        'email',
-        'profile__mail_id',
-        'profile__preferred_name',
+    queryset = Profile.objects.filter(user__email__in=emails).values(
+        'user__email',
+        'mail_id',
+        'preferred_name',
```

Comment:
The query is not substantially different from before. This is the old query:

    SELECT
      "auth_user"."email",
      "profiles_profile"."mail_id",
      "profiles_profile"."preferred_name"
    FROM "auth_user"
      LEFT OUTER JOIN "profiles_profile" ON ("auth_user"."id" = "profiles_profile"."user_id")
    WHERE "auth_user"."email" IN (c, a, b)

The new query:

    SELECT
      "auth_user"."email",
      "profiles_profile"."mail_id",
      "profiles_profile"."preferred_name"
    FROM "profiles_profile"
      INNER JOIN "auth_user" ON ("profiles_profile"."user_id" = "auth_user"."id")
    WHERE "auth_user"."email" IN (c, a, b)

The `LEFT OUTER JOIN` was changed to an `INNER JOIN` which prevents the null values from the profile table from being included. The query plans are also not substantially different:

    postgres=# explain SELECT "auth_user"."email", "profiles_profile"."mail_id", "profiles_profile"."preferred_name" FROM "auth_user" LEFT OUTER JOIN "profiles_profile" ON ("auth_user"."id" = "profiles_profile"."user_id") WHERE "auth_user"."email" IN ('c', 'a', 'b');
                                      QUERY PLAN                                   
    -------------------------------------------------------------------------------
     Hash Left Join  (cost=10.45..21.30 rows=3 width=564)
       Hash Cond: (auth_user.id = profiles_profile.user_id)
       ->  Seq Scan on auth_user  (cost=0.00..10.82 rows=3 width=520)
             Filter: ((email)::text = ANY ('{c,a,b}'::text[]))
       ->  Hash  (cost=10.20..10.20 rows=20 width=52)
             ->  Seq Scan on profiles_profile  (cost=0.00..10.20 rows=20 width=52)
    (6 rows)
    
    postgres=# explain SELECT "auth_user"."email", "profiles_profile"."mail_id", "profiles_profile"."preferred_name" FROM "profiles_profile" INNER JOIN "auth_user" ON ("profiles_profile"."user_id" = "auth_user"."id") WHERE "auth_user"."email" IN ('c', 'a', 'b');
                                      QUERY PLAN                                   
    -------------------------------------------------------------------------------
     Hash Join  (cost=10.45..21.30 rows=1 width=564)
       Hash Cond: (auth_user.id = profiles_profile.user_id)
       ->  Seq Scan on auth_user  (cost=0.00..10.82 rows=3 width=520)
             Filter: ((email)::text = ANY ('{c,a,b}'::text[]))
       ->  Hash  (cost=10.20..10.20 rows=20 width=52)
             ->  Seq Scan on profiles_profile  (cost=0.00..10.20 rows=20 width=52)
    (6 rows)